[sc-28856] Don't parse boolean operators inside other words

vkurup opened the pull request:
[Shortcut](https://app.shortcut.com/adzerk/story/28856/words-beginning-with-or-cause-zerkel-validation-error-custom-targeting)

This is a bug fix. The parser currently fails if the user searches for a variable which starts with 'and', 'not', or 'or'. [Example failure](https://adzerk.github.io/zerkel/#%7B%22zerkel%22:%22originalreferrer.raw%20contains%20%5C%22Hiring%5C%22%22,%22env%22:%22%7B%5C%22x%5C%22:%20%7B%5C%22y%5C%22:%2042%7D%7D%22%7D)

I've added tests that:
1. Show that boolean operators at the beginning of a word are not interpreted as boolean operators.
2. Show that boolean operators followed immediately by an open paren still work
3. variables are still case sensitive. This test passes on the master branch and on this branch.

The fix is a little strange. If you include just "and" in the lexical grammer, jison interprets that as a keyword all by itself. If you include ("and"), jison interprets it as a substring that it will [match even if it is not separated by spaces from the characters around it](https://github.com/zaach/jison/issues/63). Using the pipe character with "and|AND" also triggers that latter behavior. There is an [option](https://github.com/zaach/jison/issues/364) that you can provide which provides case-insensitivity for the lexer. So I change the code to use that option and then remove the pipe characters that we were using to implement case-insensitivity for the lexer.

I also improved the Makefile so that it would work on my Mac

micha reviewed (approved):
:metal: Very nice! :metal:

micha commented:
Is there any risk of a regression when the user uses a mixed case `AND`? Because before it only recognized all upper or all lower case (`AND` and `and`, respectively), now it recognizes mixed case (eg. `And`). I wonder if it's possible to now interpret a variable name like `And` as the binary operator instead of as a variable name and result in a parser exception?

vkurup commented:
Yes, excellent catch! While it makes me cringe a bit to think of using `And` as a variable, you are right that it is currently possible. So rather than cause a regression, I've removed the 'case-insensitivity' flag and added individual lexer rules for `and` and `AND`, and I've added a testcase that will catch this type of regressions in the future. (That was a confusing sentence, with 4 "and"'s in a row :D )
